### ISSUE TITLE
String nightmare in ConnectionReader/Writer

### ISSUE THREAD

**Issue by randaz81** [robotology/yarp#1953]
The following methods are problematic:
* yarp::os::ConnectionReader::expectText(const char terminatingChar = '\n')
* yarp::os::ConnectionWriter::appendString(const char* str,const char terminatingChar = '\n' )
* yarp::os::ConnectionWriter::appendRawString(const std::string& s)

Currently, the only proved way to receive a string is as: size+blob of bytes, e.g.
```
    int string_size = connection.expectInt32();
    this->map_id.resize(string_size);
    connection.expectBlock(const_cast<char*>(this->map_id.data()), string_size);
```

**Comment by drdanz** [comment]
After some investigation, I agree that this part is a nightmare. I'm planning to:

* Replace `ConnectionWriter::appendString` with `appendText` (`appendString` will be deprecated)
* Replace `ConnectionWriter::appendRawString` with `appendString` (different signature) (`appendRawString` deprecated)
* Add `ConnectionReader::expectString`
* Cleanup string serialization all around in YARP code. (This includes changing code generated by thift, but since I have a **huge** refactor on the way, I'm going to change this later).

The goal is to have either `appendText` -> `expectText` or `appendString` -> `expectString`

Since there are pure interfaces involved, this is going to be a breaking change, but I searched these methods using github search, and they seem to be used only inside YARP.

**Comment by Nicogene** [comment]
Fixed by #1972
